rootfs/etc/nginx/lua/test/balancer/sticky_test.lua

```diff
@@ -0,0 +1,151 @@
+                local default_hash_fn = "md5"
+                local default_hash_implementation = util[default_hash_fn .. "_digest"]
+                assert.equal(sticky_balancer_instance.digest_func, default_hash_implementation)
+            end)
+        end)
+    end)
+
+    describe("balance()", function()
+        setup(function()
+            util.sha1_digest = function(msg) return msg, false end -- Don't hash for simplification
+        end)
+
+        teardown(function()
+            package.loaded["util"] = nil
+            util = require("util")
+        end)
+        
+        before_each(function()
+            package.loaded["resty.chash"] = nil
+            resty_chash = require("resty.chash")
+
+            package.loaded["balancer.sticky"] = nil
+            sticky = require("balancer.sticky")
+
+            resty_chash.new = function(self, nodes)
+                return {
+                    find = function(self, key)
+                        -- cookie's value is the unhashed endpoint since we stubbed the hash function
+                        assert.equal(key, test_backend_endpoint)
+                        return test_backend_endpoint
+                    end,
+                    next = function(self, index)
+                        return test_backend_endpoint
+                    end,
+                    npoints = 1
+                }
+            end
+        end)
+
+        context("when client doesn't have a cookie set", function()
+            it("should pick an endpoint for the client", function()
+                local cookie = require("resty.cookie")
+                cookie.new = function(self) 
+                    return { 
+                        get = function(self, n) return false end,
+                        set = function(self, n) return true end
+                    }
+                end
+                local sticky_balancer_instance = sticky:new(test_backend)
+                local s = spy.on(sticky_balancer_instance.instance, "next")
+                assert.has_no.errors(function() sticky_balancer_instance:balance() end)
+                assert.spy(s).was_called()
+            end)
+
+            it("should set a cookie on the client", function() 
+                local cookie = require("resty.cookie")
+                local s = {}
+                cookie.new = function(self) 
+                    local cookie_instance = {
+                        set = function(v) return true, nil end,
+                        get = function(k) end,
+                    }
+                    s = spy.on(cookie_instance, "set")
+                    return cookie_instance, false
+                end
+                local sticky_balancer_instance = sticky:new(get_test_backend())
+                assert.has_no.errors(function() sticky_balancer_instance:balance() end)
+                assert.spy(s).was_called()
+            end)
+        end)
+
+        context("when client has a cookie set", function()
```

Comment:
You are missing a crucial test here. We should make sure `:balance` returns the correct endpoint for given cookie value.